### ISSUE TITLE
Microsoft Teams notification support

### ISSUE THREAD

**Issue by linitfor** [SamsidParty/TopNotify#7]
As title. The MS Teams notification cannot support.

**Comment by linitfor** [comment]
teams work/school 
![圖片](https://github.com/SamsidParty/TopNotify/assets/58168507/7e1a6329-710d-4a98-a37e-b82d69dba342)
![圖片](https://github.com/SamsidParty/TopNotify/assets/58168507/5a1a9713-4529-4d0f-a0b9-0eb18a90240d)

**Comment by SamarthMP** [comment]
Alright, teams work/school uses custom notifications. When I get home I will try to add an interceptor for it.

**Comment by linitfor** [comment]
thanks ~

**Comment by linitfor** [comment]
> Alright, teams work/school uses custom notifications. When I get home I will try to add an interceptor for it.

Have you success to implement this work ?

**Comment by SamsidParty** [comment]
Still working on it, teams is a beast that I have to battle with. You can expect a stable version in the next week or so.

**Comment by SamsidParty** [comment]
Unfortunately, we don't have a work/school account to use on teams. MS is soon going to roll out support for personal accounts on the new teams app. Until then, it's not possible for me to implement support. Feel free to open a pull request though.

**Comment by SamsidParty** [comment]
Support is coming in the next release.

**Comment by SamsidParty** [comment]
Hopefully in the next week. The interception method for teams is completely different. We have to inject a dll into teams that overrides the SetWindowPos function. Not sure if the MS store will allow that.

**Comment by SamarthMP** [comment]
Development is going well, still have to squash some bugs and prepare for publishing to the MS Store.
A beta release will be created within the next few days if you want to test the feature.
![image](https://github.com/SamsidParty/TopNotify/assets/79756986/49028278-486b-42e0-a8c7-92dcb2f363ee)

**Comment by SamarthMP** [comment]
A beta release has been published with this feature.
You'll have to uninstall the current TopNotify and install a CA certificate, since it's not from the MS Store.

**Comment by bel57** [comment]
Oh boy, I can't wait for that feature to be deployed in the MS Store version.
The fact that you can't disable notifications for specific posts in Teams is really dumb, at least with this tool I could see through the spamming rectangle.

Please support 'new' Teams, currently v49/24051622223 (build 24152.412.2958.9166 updated today).

**Comment by SamsidParty** [comment]
Interesting, it should be working on version 49/24050313444 with the "new" teams.
Did you guys install teams from the website or the MS store?

**Comment by SamsidParty** [comment]
Alright, i've looked into it, and found out that it doesn't work with the old teams. The "new" teams you have @linitfor is actually the old teams with a toggle switch into new teams mode. Microsoft has made this very complex for no reason. If you don't mind, can you please uninstall the teams you have and install it from [here](https://www.microsoft.com/en-us/microsoft-teams/download-app) under download the new teams app.

The real new teams app is an MSIX file and does not have a toggle switch, that one works with TopNotify.

**Comment by bel57** [comment]
I do have the new Teams. Made sure to uninstall old version, no toggle anymore, and there's the new splash screen + logo.
Can install some MS apps and software via 3rd party via MS Store.
Can't try the beta ok my workstation tho, will have to wait for the update to be pushed to the store.

**Comment by SamarthMP** [comment]
I used [this](https://statics.teams.cdn.office.net/production-windows-x64/enterprise/webview2/lkg/MSTeams-x64.msix) MSIX file from their website.

**Comment by SamsidParty** [comment]
Can you confirm if it's working? I would like to have confirmation before I push an update to the MS store.

**Comment by SamsidParty** [comment]
I will not be home until July 10th. Thanks for your patience.

**Comment by SamsidParty** [comment]
Alright, turns out it only worked when display scaling was set to 100%. I will try to provide a fix ASAP.

**Comment by SamsidParty** [comment]
Added in version 2.4.1, please update TopNotify from the MS Store.

**Comment by bel57** [comment]
Doesn't work on my workstation with new Teams installed, unfortunately.

**Comment by SamsidParty** [comment]
Can you please upload your daemon.ivlog file?

Steps:
- Press F2 on the settings panel
- Open The App Data Folder
- Close TopNotify Using The Tray Icon
- Upload daemon.ivlog

**Comment by SamsidParty** [comment]
It seems like the failure point is the teams injection. A common cause for this is if you have the 32 bit version of teams.
In task manager (details section), does ms-teams.exe show x64 or x86?
Also, as discussed previously, does teams have a "New Teams" switch in the menu?

**Comment by bel57** [comment]
New Teams, 64 bits, no prompt/switch button as I'm already running the latest version.
![image (1)](https://github.com/user-attachments/assets/e1d8618c-5411-40b8-b65b-0616e330bee3)

**Comment by SamsidParty** [comment]
Interesting, I have an identical teams version. There are only 2 ms-teams.exe processes running right?

**Comment by bel57** [comment]
Yep these are the only 2 Teams processes. Under Teams there are also 10 Edge Webview2 and a Runtime Broker listed.

**Comment by SamsidParty** [comment]
I will have to look into it, I may switch from EasyHook to Detours considering how many problems it is causing.

**Comment by SamsidParty** [comment]
New updates for Teams and TopNotify have been released, it it working now?

**Comment by SamsidParty** [comment]
I see, well version 2.4.4 has some misc optimizations for the MS Teams interceptor, maybe that'll work when it releases.

**Comment by SamsidParty** [comment]
2.4.4 is out, if it's still not working then looks like we will have to do more troubleshooting.

**Comment by SamsidParty** [comment]
I have scrapped teams support because it has caused nothing but problems in TopNotify. Microsoft is still working on their rewrite of teams, I may come back to it in the future when it's more stable.
